Packages/ChatSecure/ChatSecure/Classes/Networking/Services/Messsage/MessageService.swift

```diff
@@ -8,134 +8,338 @@
+			request.senderKeyID = Int64(distributionId.uuidString) ?? 0
+			
+			guard let server = channelStorage.realmManager.getServer(by: domain) else { return false }
+			let response = await channelStorage.getChannel(domain: server.serverDomain, accessToken: server.accessKey, hashKey: server.hashKey).groupRegisterClientKey(request)
+			switch response {
+			case .success(let result):
+				return true
+			case .failure(let error):
+				return false
+			}
+		} catch {
 			return false
 		}
 	}
 	
-	func processKeyStoreHasPrivateKey(recipientResponse: Signal_PeerGetClientKeyResponse) {
-		if let ourEncryptionMng = CKSignalCoordinate.shared.ourEncryptionManager {
+	func decryptPeerMessage(senderName: String, message: Data) -> String? {
+		do {
+			let signalProtocolAddress = try ProtocolAddress(name: senderName, deviceId: UInt32(Constants.receiverDeviceId))
+			let preKeyMessage = try PreKeySignalMessage(bytes: message)
+			let decryptMessage = try signalDecryptPreKey(
+				message: preKeyMessage, from: signalProtocolAddress, sessionStore:
+				signalStore.sessionStore, identityStore: signalStore.identityStore,
+				preKeyStore: signalStore.preKeyStore,
+				signedPreKeyStore: signalStore.signedPreKeyStore, context: NullContext())
+			return String(bytes: decryptMessage, encoding: .utf8)
+		} catch {
+			return nil
+		}
+	}
+	
+	func decryptGroupMessage(senderId: String, senderDomain: String, ownerId: String, ownerDomain: String, groupID: Int64, message: Data) async -> String? {
+		do {
+			let senderAddress: ProtocolAddress
+			let distributionId: UUID
+			if senderId == ownerId {
+				senderAddress = try ProtocolAddress (name: "\(senderDomain)_\(senderId)", deviceId: UInt32(Constants.senderDeviceId))
+				distributionId = senderStore.getSenderDistributionID(from: senderAddress, groupId: groupID)
+			} else {
+				senderAddress = try ProtocolAddress (name: "\(senderDomain)_\(senderId)", deviceId: UInt32(Constants.receiverDeviceId))
+				distributionId = senderStore.getSenderDistributionID(from: senderAddress, groupId: groupID)
+			}
+			await initSessionUserInGroup(address: senderAddress, distributionId:
+											distributionId, domain: ownerDomain, groupID: groupID,
+										 clientID: senderId, isForceProcess: false)
 			do {
-				let remotePrekey = try SignalPreKey(serializedData: recipientResponse.preKey)
-				let remoteSignedPrekey = try SignalSignedPreKey(serializedData: recipientResponse.signedPreKey)
-				
-				guard let preKeyKeyPair = remotePrekey.keyPair,
-					  let signedPrekeyKeyPair = remoteSignedPrekey.keyPair else {
-					return
+				let plaintextFromAlice = try groupDecrypt(message, from: senderAddress, store: senderStore, context: NullContext())
+				return String(bytes: plaintextFromAlice, encoding: .utf8)
+			} catch {
+				let initSessionAgain = await initSessionUserInGroup(address: senderAddress, distributionId:
+																		distributionId, domain: ownerDomain, groupID: groupID,
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ClearKeep/ck-ios/blob/88c1f4f9b0f08387d7d5bf2f7e72ae38123bc3b5/Packages/ChatSecure/ChatSecure/Classes/Networking/Services/Messsage/MessageService.swift#L295"></span>Arguments should be either on the same line, or one per line.<br />
<code>multiline_arguments</code> <code>MessageService.swift:295</code></td>
    </tr>
  </tbody>
</table>